dags/maxtext_pathways/configs/utils.py

```
@@ -188,3 +275,237 @@ def check_gcp_logs_exist(
+  contain a specific substring.
+  """
+  _, pods = authenticate_kubectl(project_id, region, cluster_name, workload_id)
+
+  if not pods.items:
+    logging.info("No pods found for workload selector: %s.", workload_id)
+    return False
+
+  for pod in pods.items:
+    if pod.status.phase == "Failed":
+      # Don't keep retrying if the pod has failed
+      raise AirflowFailException(f"Bad pod phase: {pod.status.phase}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)